src/components/Explore/LogsByService/Tabs/GenericBreakdownScene.tsx

```diff
@@ -0,0 +1,39 @@
+import { SceneCSSGridItem, SceneObjectBase, SceneObjectState, SceneQueryRunner } from '@grafana/scenes';
+import { ALL_VARIABLE_VALUE } from '../../../../utils/shared';
+import { LoadingState, SelectableValue } from '@grafana/data';
+
+export interface GenericBreakdownSceneState extends SceneObjectState {
+  labels: Array<SelectableValue<string>>;
+  changeLabels?: (n: string[]) => void;
+}
+
+export class GenericBreakdownScene<
+  TState extends SceneObjectState = SceneObjectState
+> extends SceneObjectBase<GenericBreakdownSceneState> {
+  protected removeErrorsAndSingleCardinality(queryRunner: SceneQueryRunner, gridItem: SceneCSSGridItem) {
+    queryRunner.getResultsStream().subscribe((result) => {
+      if (result.data.errors && result.data.errors.length > 0) {
+        const val = result.data.errors[0].refId!;
+        this.hideField(val);
+        gridItem.setState({ isHidden: true });
+      } else if (result.data.state === LoadingState.Done) {
+        // Hide panels with single cardinality
+        if (result.data.series.length < 2) {
+          const val = result.data.series?.[0]?.refId;
+          gridItem.setState({ isHidden: true });
+
+          if (val) {
+            this.hideField(val);
+          }
```

Comment:
Still seeing No Labels for detected fields 
![Screenshot 2024-04-12 at 11 40 39 AM](https://github.com/grafana/explore-logs/assets/114438185/b0124066-3303-4a81-a4ef-046f07a44ea8) I'd think the only change would need to be in the data response here and the rest could be left as is. I wonder if the label count doesn't change because they are there just hidden. @sven did something like this before I thought where he just remove data from the response  instead of hiding`const detectedFields = res.fields.filter((f) => !disabledFields.includes(f)).sort((a, b) => a.localeCompare(b));`

Reply:
Yeah actually that would be a much better approach in general, we could just remove single cardinality/error dataframes at the top level after we do the extractFields transformation. That would fix some UX weirdness too with the graphs disappearing on load.

I'll close this and try that approach on monday! Thank you @L2D2Grafana

Reply:
So that won't work as we found out in https://github.com/grafana/explore-logs/pull/226, the labels from the logs panel dataframe are only the labels of the log lines, not the complete set of labels for the query. So if we calculate the cardinality before we execute each sub-query aggregated by each label, we could be wrong, particularly in cases where the query result is a tiny subset